### PR TITLE
Implement sharing functionality for Enterprise

### PR DESCRIPTION
**Pull request by mmargarit**
The following payload can trigger the share 

`var payload = {"text":"Text to share", "url": "https://getdreams.com"};
if (window.JSBridge) {
    window.JSBridge.share(payload);
}`

### CODE REVIEW

File: sdk/src/androidTest/java/com/getdreams/views/DreamsViewTest.kt [hunk 1]
```diff
@@ -293,4 +293,20 @@ class DreamsViewTest {
...
+        server.dispatcher = MockDreamsDispatcher(server)
+        server.start()
+        Dreams.configure(Dreams.Configuration("clientId", server.url("/").toString()))
+
+        val latch = CountDownLatch(1)
+        activityRule.testResponseEvent("share_button") { event, _ ->
+            assertEquals(Event.Share, event)
+            latch.countDown()
+        }
+       assertTrue(latch.await(5, TimeUnit.SECONDS))
```
Comment (ninaniiranen) [inline sdk/src/androidTest/java/com/getdreams/views/DreamsViewTest.kt]: We should verify that the share intent was sent.

File: sdk/src/main/java/com/getdreams/events/Event.kt [hunk 2]
```diff
@@ -31,4 +31,9 @@ sealed class Event {
      * Event sent when the Dreams should be exited.
      */
     object ExitRequested : Event()
+
+    /**
+     * Event sent when share should be triggered in Dreams .
```
Comment (ninaniiranen) [inline sdk/src/main/java/com/getdreams/events/Event.kt]: Rouge space before the full stop punctuation.

File: sdk/src/main/java/com/getdreams/connections/webview/ResponseInterface.kt [hunk 3]
```diff
@@ -31,4 +31,9 @@ internal interface ResponseInterface {
      * The user requested to exit Dreams.
      */
     fun onExitRequested()
+
+    /**
+     * The user requests share dialog to be prompted
```
Comment (ninaniiranen) [inline sdk/src/main/java/com/getdreams/connections/webview/ResponseInterface.kt]: Missing full stop.

**Review by ninaniiranen — CHANGES_REQUESTED** [review]
Some nitpicking for the comments, but we should verify that the intent was sent.
Also the commit messages should follow https://www.conventionalcommits.org/en/v1.0.0/ so that we can auto-generate changelogs with https://github.com/conventional-changelog/standard-version#cli-usage.

**Comment by mmargarit** [comment]
haha I missed klint format here :) 
Fixed linting issues here , will rneed to esearch how to verify the intent ..
Thanks for the tip about commints..,didn't know 👍

**Comment by mmargarit** [comment]
@ninaniiranen Added hits check with intent filter, tried to check intent content too but weirdly sometimes it gets them properly sometimes fails with null, I assume because of timeouts. Do you think check with hits is enough ?

**Review by ninaniiranen — APPROVED** [review]

**Pull request merged**